src/lib.rs

```diff
@@ -384,6 +415,381 @@ macro_rules! soa_zip {
+            self.get_mut(self.len().saturating_sub(1))
+        }
+
+        /// Obtain a `mut` pointer type for this data
+        fn as_mut_ptr(&mut self) -> Self::PtrMut;
+
+        /// Create a new, empty struct of arrays
+        fn new() -> Self;
+
+        /// Create a new, empty struct of arrays with the specified capacity
+        fn with_capacity(capacity: usize) -> Self;
+
+        /// Analogous to [`Vec::capacity`]
+        fn capacity(&self) -> usize;
+
+        /// Analogous to [`Vec::reserve`]
+        fn reserve(&mut self, additional: usize);
+
+        /// Analogous to [`Vec::reserve_exact`]
+        fn reserve_exact(&mut self, additional: usize);
+
+        /// Analogous to [`Vec::shrink_to_fit`]
+        fn shrink_to_fit(&mut self);
+
+        /// Analogous to [`Vec::truncate`]
+        fn truncate(&mut self, len: usize);
+
+        /// Add a singular value of `T` to the arrays. Analogous to [`Vec::push`]
+        fn push(&mut self, value: T);
+
+        /// Analogous to [`Vec::swap_remove`]
+        fn swap_remove(&mut self, index: usize) -> T;
+
+        /// Analogous to [`Vec::insert`]
+        fn insert(&mut self, index: usize, element: T);
+
+        /// Similar to [`std::mem::replace()`](https://doc.rust-lang.org/std/mem/fn.replace.html).
+        fn replace(&mut self, index: usize, element: T) -> T;
+
+        /// Analogous to [`Vec::remove`]
+        fn remove(&mut self, index: usize) -> T;
+
+        /// Analogous to [`Vec::pop`]
+        fn pop(&mut self) -> Option<T>;
+
+        /// Analogous to [`Vec::append`]
+        fn append(&mut self, other: &mut Self);
+
+        /// Analogous to [`Vec::clear`]
+        fn clear(&mut self);
+
+        /// Analogous to [`Vec::split_off`]
+        fn split_off(&mut self, at: usize) -> Self;
+    }
+
+    /// A trait to implement `Clone`-dependent behavior to convert a non-owning SoA type into an
+    /// owning [`SoAVec`].
+    pub trait ToSoAVec<T: StructOfArray> {
+        type SoAVecType: SoAVec<T>;
+
+        /// Similar to [`slice::to_vec()`](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec)
+        fn to_vec(&self) -> Self::SoAVecType;
+    }
+
+    /// A trait to implement `Clone`-dependent behavior to extend an [`SoAVec`] with data copied
+    /// from its associated `Slice` type.
+    pub trait SoAAppendVec<T: StructOfArray>: SoAVec<T> {
+
+        /// Analogous to [`Vec::extend_from_slice`]
+        fn extend_from_slice(&mut self, other: Self::Slice<'_>);
+    }
+
+    /// A trait to express the [`IntoIterator`] guarantee of [`SoASlice`] types in the type system.
+    pub trait IntoSoAIter<'a, T: StructOfArray>: SoASlice<T> + IntoIterator<Item=Self::Ref<'a>> + 'a {}
+}
+pub use generics::*;
+
+/// A collection of supporting traits for [`StructOfArray`] bundled in one place for ease-of-access
+pub mod prelude {
+    pub use super::{SoAVec, SoAIter, SoASlice, SoASliceMut, SoAPointers, StructOfArray, ToSoAVec, IntoSoAIter, SoAAppendVec};
+}
```

Comment:
This is everything defined by this crate, isn't it? I don't think it's worth having a separate prelude in this case, the same can be achieved with `use soa_derive::*` instead of `use soa_derive::prelude::*`

Reply:
That'd also pull the `soa_zip` macro and its helper, `soa_zip_impl`, along with `SoAIndex`, which isn't necessary to use the generic traits or the types themselves. 

The `prelude` also means that if in the future you were to add more things to the crate root, they would only be pulled by a wildcard import if they were also added to the prelude. This is probably overly defensive future-proofing though.

If you still think this is acceptable, I'm fine with removing the `prelude` module.